Add Weave Scope

errordeveloper opened the pull request.

errordeveloper commented:
Would be great if I could specify license and URLs for product info and bug reports.

ibuildthecloud commented:
Additionally a maintainer field and a long form README would be great.

@prachidamle @vincent99

cloudnautique commented:
Mentioned it to @lvuch as well.

errordeveloper commented:
@ibuildthecloud ping

johnrengelman commented:
I actually ran into some issues with this because using `net: host` causes the container to not be able to be load balanced by the Rancher LB. So I ended up going with 2 containers:
1. The global container as specified here, but passing the `--no-app` flag to it
2. A single container that runs with the regular networking and has the `--no-probe` flag passed to it.

I then linked the single container into the global containers so they would send their reports to the singular app container which could then be load balanced.

```
weavescope-probes:
  image: weaveworks/scope:0.9.0
  privileged: true
  net: host
  pid: host
  labels:
    io.rancher.scheduler.global: true
    io.rancher.container.dns: true
  links:
    - weavescope-apps:weavescope
  ports:
  - "4040:4040"
  volumes:
  - "/var/run/docker.sock:/var/run/docker.sock"
  tty: true
  command:
  - "--probe.docker"
  - "true"
  - "--no-app"
  - "weavescope"
weavescope-apps:
  image: weaveworks/scope:0.9.0
  privileged: true
  volumes:
  - "/var/run/docker.sock:/var/run/docker.sock"
  command:
  - "--no-probe"
```

errordeveloper commented:
@johnrengelman thanks for looking into it! @tomwilkie @peterbourgon @paulbellamy what do you folks think about this? Also, we probably need to bump the version.

peterbourgon commented:
Just took a quick readthru, but LGTM. The latest stable is indeed 0.10.0 and comes with some nice control features, and we'll push 0.11.0 before Christmas, but 0.9.0 is a fine version too :)

errordeveloper commented:
@peterbourgon thanks for chiming in. Do you have any thoughts on @johnrengelman's solution with `--no-app`/`--no-probe`?

peterbourgon commented:
Yes, it seems like a reasonable split to me.

errordeveloper commented:
Updated per discussion and rebased.

pidster commented:
@errordeveloper update to v0.11.x pls.

errordeveloper commented:
I've rebased and updated this to use latest version of Scope. @ibuildthecloud is there anything else for us to update? ...I can squash, if you really want me to :)

errordeveloper commented:
I assume that test failures have nothing to do with the change here, is that correct or I'm just not able to read the output?

cloudnautique commented:
It looks like the service might not have been up yet in ci. Will need to test when I have fast internets again.

cloudnautique commented:
While we wait :) we added support for README.md files either in the numbered folder or in the same dir as the icon. Take a look at the Hadoop or zookeeper entries for examples if you are interested.

cloudnautique commented:
The CI blowing up is a real issue. The catalog service requires a rancher-compose.yml file. 

at a minimum it needs to have:

```
.catalog:
  name: something
  version: x.x
  description: |
    some thing
  uuid: weavescope-0
```

That should get what it needs. The version here should line up with the version in the config.yml and the UUID just needs to be something unique.

errordeveloper commented:
It's looking all good now :+1:

errordeveloper commented:
Do you need anything else from me or it can be merged as is?

On Fri, 18 Dec 2015 20:50 Bill Maxwell notifications@github.com wrote:

> The CI blowing up is a real issue. The catalog service requires a
> rancher-compose.yml file.
> 
> at a minimum it needs to have:
> 
> .catalog:
>   name: something
>   version: x.x
>   description: |
>     some thing
>   uuid: weavescope-0
> 
> That should get what it needs. The version here should line up with the
> version in the config.yml and the UUID just needs to be something unique.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/rancher/rancher-catalog/pull/26#issuecomment-165895101
> .

errordeveloper commented on templates/weavescope/0/docker-compose.yml:
```diff
@@ -0,0 +1,27 @@
+  - "--probe.docker"
+  - "true"
+  - "--no-app"
+  - "weavescope"
```
It will be looking for `weavescope` in DNS, how is that supposed to work now, since we separated the app and the probes? ...and is the `weavescope-apps:weavescope` link correct in this context?

cloudnautique replied:
It will get the DNS entry because of the link. There will be a slight lag of few seconds while the resolv.conf is setup inside the container, will the app die if it can't resolve? or will it retry? 

Also, yes please squash :)

errordeveloper replied:
It won't die, it'll poll periodically. Also, you can get rid of the link and just pass `weavescope-apps` instead of `weavescope`.

errordeveloper commented:
@cloudnautique how can I test or demo this before it gets merged?

cloudnautique commented:
Yes, you need a copy of the repo locally or hosted someplace with your changes on master. It just needs to be able to clone it so it can be local filesystem or remote.

Then you need to setup a rancher env. When deploying [Rancher Server](http://docs.rancher.com/rancher/installing-rancher/installing-server/)launch an instance of rancher/server add this `-e CATTLE_CATALOG_URL=<your repo>` if your using the local filesystem as your repo also do a bind mount into the container `-v <path_to_your_repo>:/catalog` or something like that.

Then add a host to the env so you can try deploying... probably want 2 nodes to test the networking.

rlex commented:
Works OK for me with 2 nodes, but i used rancher-compose to deploy it.

errordeveloper commented:
@cloudnautique thanks! I'm working on this right now and should have it finished and fully tested shortly. One question regarding `CATTLE_CATALOG_URL` - is it only cloned once and never pulled again?

cloudnautique commented:
Its pulled every 10 minutes or so. Its actually a lot easier then this now. In rancher/server:v0.51.0 we added support for multiple catalogs. So if you have this PR on master in your own repo, just add it in the admin > settings tab in the Rancher UI. No need to mess with env vars :)

errordeveloper commented:
@cloudnautique ah, I see. I'll try that.

errordeveloper commented:
@cloudnautique I have another issue here, which I just mentioned to @ibuildthecloud on IRC... Looks like `weavescope-probe` (which has to be in host's network namespaces) is not able to ping or connect to `weavescope-app`, unless  `weavescope-app` is on the same host. Any ideas how to fix this?

cloudnautique commented:
Is it by DNS name? If you want to resolve other Rancher services in host network mode, you need to add the label `io.rancher.container.dns: "true"`

errordeveloper commented:
I have `io.rancher.container.dns: "true"` set already and resolution actually works, but pings don't.

cloudnautique commented:
![screen shot 2016-01-25 at 12 04 51 pm](https://cloud.githubusercontent.com/assets/1989084/12561181/ea602976-c35b-11e5-83bb-5e3ad71cda8e.png)

errordeveloper committed:
Add Weave Scope v0.11.1

errordeveloper commented on templates/weavescope/0/docker-compose.yml:
```diff
@@ -0,0 +1,24 @@
+  labels:
+    io.rancher.scheduler.global: true
+    io.rancher.container.dns: true
+  links:
```
@cloudnautique can you test it without the link? ...I'd expect it to work without it.

cloudnautique replied:
Tested, and it works without the link.

cloudnautique commented:
LGTM
